File: python/infinicore/ops/matmul.py
from infinicore.lib import _infinicore
from infinicore.tensor import Tensor


def matmul(input, other, *, out=None):
    if out is None:
        return Tensor(_infinicore.matmul(input._underlying, other._underlying))

    _infinicore.matmul_(out._underlying, input._underlying, other._underlying)

File: src/infinicore/ops/matmul/matmul.cc
#include "infinicore/ops/matmul.hpp"
#include "infinicore/ops/gemm.hpp"

namespace infinicore::op {

Tensor matmul(Tensor a, Tensor b) {
    return gemm(a, b, 1.0f, 0.0f);
}

void matmul_(Tensor c, Tensor a, Tensor b) {
    Gemm::execute(c, a, b, 1.0f, 0.0f);
}
} // namespace infinicore::op
